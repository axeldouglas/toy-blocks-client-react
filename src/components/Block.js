import React from "react";
import PropTypes from "prop-types";
import { Typography, makeStyles, Box } from "@material-ui/core";
import colors from "../constants/colors";

const Block = ({ block }) => {
  const classes = useStyles();

  return (
    <Box className={classes.blockContainer}>
      <Typography className={classes.blockTitle}>
        {block?.id?.padStart(3, "0") || "000"}
      </Typography>
      <Typography className={classes.blockDescription}>
        {block?.attributes?.data || "---"}
      </Typography>
    </Box>
  );
};

const useStyles = makeStyles((theme) => ({
  blockContainer: {
    display: "flex",
    flexDirection: "column",
    backgroundColor: colors.disabled,
    borderRadius: "2px",
    padding: "8px 8px 4px 8px",
    margin: "2px 0",
    fontStyle: "normal",
  },
  blockTitle: {
    fontWeight: "bold",
    fontSize: theme.typography.pxToRem(10),
    lineHeight: "16px",
    letterSpacing: "1.5px",
    textTransform: "uppercase",
    color: colors.primary,
  },
  blockDescription: {
    fontWeight: "normal",
    fontSize: theme.typography.pxToRem(14),
    lineHeight: "20px",
    letterSpacing: "0.25px",
    color: colors.text,
  },
}));

Block.propTypes = {
  block: PropTypes.shape({
    id: PropTypes.string,
    type: PropTypes.string,
    attributes: PropTypes.shape({
      index: PropTypes.number,
      timestamp: PropTypes.number,
      data: PropTypes.string,
      "previous-hash": PropTypes.string,
      hash: PropTypes.string,
    }),
  }).isRequired,
};

export default Block;
