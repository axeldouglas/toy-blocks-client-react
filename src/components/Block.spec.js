import React from "react";
import { create } from "react-test-renderer";
import Block from "./Block";

describe("<Block />", () => {
  const block = {
    id: "5",
    type: "blocks",
    attributes: {
      index: 1,
      timestamp: 1530679678,
      data: "The Human",
      "previous-hash": "IoVe8=",
      hash: "oHkxOJW=",
    },
  };

  const blockEmpty = {
    attributes: {
      data: "",
    },
  };

  it("should match empty data snapshot", () => {
    const component = create(<Block block={blockEmpty} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot", () => {
    const component = create(<Block block={block} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
