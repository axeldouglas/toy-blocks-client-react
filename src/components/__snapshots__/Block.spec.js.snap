// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Block /> should match empty data snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-4 makeStyles-blockContainer-1"
>
  <p
    className="MuiTypography-root makeStyles-blockTitle-2 MuiTypography-body1"
  >
    000
  </p>
  <p
    className="MuiTypography-root makeStyles-blockDescription-3 MuiTypography-body1"
  >
    ---
  </p>
</div>
`;

exports[`<Block /> should match snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-5 makeStyles-blockContainer-1"
>
  <p
    className="MuiTypography-root makeStyles-blockTitle-2 MuiTypography-body1"
  >
    005
  </p>
  <p
    className="MuiTypography-root makeStyles-blockDescription-3 MuiTypography-body1"
  >
    The Human
  </p>
</div>
`;
